File: chroma-league/src/test/java/com/bonepl/chromaleague/tasks/CheckRiotApiTaskTest.java
package com.bonepl.chromaleague.tasks;

import com.bonepl.chromaleague.state.RunningState;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertFalse;

class CheckRiotApiTaskTest {

    @Test
    void shouldFailWhenRiotApiNotUp() {
        //given
        RunningState.getRiotApi().setValue(true);

        //when
        new CheckRiotApiTask().run();

        //then
        assertFalse(RunningState.getRiotApi().getValue());
    }
}

File: test/tasks/CheckRiotApiTask.test.ts
import {beforeEach, describe, expect, test, vi} from 'vitest';
import {RunningState} from '../../src/state/RunningState.js';

vi.mock('../../src/tasks/RiotApiChecker.js', () => ({
    checkRiotApiUp: vi.fn().mockResolvedValue(false),
}));

import { CheckRiotApiTask } from '../../src/tasks/CheckRiotApiTask.js';

describe('CheckRiotApiTask', () => {
    beforeEach(() => {
        RunningState.setRiotApi(true);
        RunningState.setRunningGame(false);
    });

    test('shouldFailWhenRiotApiNotUp', async () => {
        const task = new CheckRiotApiTask();

        //when - 3 consecutive failures required before disconnect
        await task.run();
        await task.run();
        await task.run();

        //then
        expect(RunningState.getRiotApi().getValue()).toBe(false);
    });
});
